client/src/Components/Layout/SignUp/SignUp.jsx

```diff
@@ -0,0 +1,597 @@
+import React, { Component } from 'react';
+import { Link, Redirect } from 'react-router-dom';
+import DateFnsUtils from '@date-io/date-fns';
+import { withStyles } from '@material-ui/core/styles';
+import LoaderProgress from '../../Common/LoaderProgress';
+
+import {
+  Box,
+  Grid,
+  TextField,
+  Button,
+  Typography,
+  Paper,
+  FormHelperText,
+  FormControl,
+  Input,
+  InputLabel,
+  IconButton,
+  InputAdornment,
+  s,
+} from '@material-ui/core';
+
+import {
+  Person,
+  AlternateEmail,
+  Phone,
+  Lock,
+  DateRange,
+  Visibility,
+  VisibilityOff,
+} from '@material-ui/icons';
+
+import { MuiPickersUtilsProvider, DatePicker } from '@material-ui/pickers';
+
+import {
+  validateEmail,
+  stringValidation,
+  phoneValidation,
+} from '../../../helpers/Validation';
+
+import queryString from 'query-string';
+import Cookies from 'js-cookie';
+import SignUpStyle from './SignUpStyle';
+
+class SignUp extends Component {
+  state = {
+    userDetails: {
+      name: {
+        value: '',
+        message: '',
+        isValid: true,
+        isRequired: true,
+      },
+      phone: {
+        value: '',
+        message: '',
+        isValid: true,
+        isRequired: true,
+      },
+      email: {
+        value: '',
+        message: '',
+        isValid: true,
+        isRequired: true,
+      },
+      password: {
+        value: '',
+        message: '',
+        isValid: true,
+        isRequired: true,
+      },
+      rePassword: {
+        value: '',
+        message: '',
+        isValid: true,
+        isRequired: true,
+      },
+      showPassword: false,
+      showRePassword: false,
+      returnUrlText: '/',
+      redirect: false,
+      loginLink: '/user/login',
+      birthDate: '',
+      minDate: '',
+      maxDate: '',
+    },
+  };
+
+  componentDidMount() {
+    // max age of member must be less than 100 year
+    // actually Do not happen , but it type of validation
+    const min = new Date();
+    min.setDate(1);
+    min.setMonth(0);
+    min.setFullYear(min.getFullYear() - 100);
+
+    // the member age at least 4 years
+    const max = new Date();
+    max.setFullYear(max.getFullYear() - 4);
+
+    const AuthToken = Cookies.get('AuthToken');
+    const qstring = queryString.parse(this.props.history.location.search);
+    if (qstring.ReturnUrl) {
+      this.setState({
+        returnUrlText: qstring.ReturnUrl,
+        loginLink: `/user/login/?ReturnUrl=${qstring.ReturnUrl}`,
+        minDate: min,
+        maxDate: max,
+        birthDate: min,
+      });
+    } else {
+      this.setState({
+        // if the user log in we will redirect to Home
+        redirect: !AuthToken ? false : true,
+        minDate: min,
+        maxDate: max,
+        birthDate: min,
+      });
+    }
+  }
+
+  handleClickShowPassword = () => {
+    const showPassword = this.state.showPassword;
+
+    this.setState({
+      showPassword: !showPassword,
+    });
+  };
+
+  handleMouseDownRePassword = (event) => {
```

Comment:
I'm curious, what are these mousedown handlers for?

Reply:
it for icon in password field